patches/tModLoader/Terraria/GameContent/UI/Elements/UIList.cs.patch

```diff
@@ -24,15 +32,31 @@
  		public virtual void Add(UIElement item) {
  			_items.Add(item);
  			_innerList.Append(item);
-+			UpdateOrder();
-+			_innerList.Recalculate();
-+		}
-+
+@@ -72,6 +_,15 @@
+ 			_innerList.Recalculate();
+ 		}
+ 
 +		public virtual void AddRange(IEnumerable<UIElement> items) {
 +			_items.AddRange(items);
 +			foreach (var item in items)
 +				_innerList.Append(item);
 +
++			UpdateOrder();
++			_innerList.Recalculate();
++		}
++
+ 		public virtual bool Remove(UIElement item) {
+ 			_innerList.RemoveChild(item);
  			UpdateOrder();
- 			_innerList.Recalculate();
+@@ -86,6 +_,11 @@
+ 		public override void Recalculate() {
+ 			base.Recalculate();
+ 			UpdateScrollbar();
++		}
++
++		public override void MouseOver(UIMouseEvent evt) {
++			base.MouseOver(evt);
++			PlayerInput.LockVanillaMouseScroll();
  		}
```

Comment:
This, AddRange, and ViewPosition should go into a partial file.

Reply:
The following isn't your change, but can you check if there's any reason why _items is public and not internal?

Reply:
Regards to the latter, it was changed back in 2016 to be accessible for the ModBrowser; I suspect there wasn't an observed vanilla method to access the individual elements of the list. 
There is a fetchable enumerator now, so I feel comfortable changing to internal.